ogr/services/github.py

```diff
@@ -30,23 +30,22 @@
 class GithubService(BaseGitService):
     # class parameter could be use to mock Github class api
     github_class: Type[github.Github]
+    persistent_storage: Optional[PersistentObjectStorage] = None
 
     def __init__(
         self,
         token=None,
         read_only=False,
-        persistent_storage_file=None,
-        is_persistent_storage_write_mode=False,
+        persistent_storage: Optional[PersistentObjectStorage] = None,
     ):
         super().__init__()
         self._token = token
-        if not hasattr(self, "github_class"):
-            if persistent_storage_file:
-                self.github_class = get_Github_class(
-                    persistent_storage_file, is_persistent_storage_write_mode
-                )
-            else:
-                self.github_class = github.Github
+        if persistent_storage:
+            self.persistent_storage = persistent_storage
```

Comment:
You can avoid the `if persisent_storage` and use directly the `self.persistent_storage = persistent_storage`.

Reply:
@lachmanfrantisek No, it is not possible,
because `persistent_storage` is None by default, and I expect that tests for `packit` set this class attribute and will not pass param directly.

Reply:
OK, it is a little confusing that it is a class attribute but we can leave it as is for now.

Reply:
yep, I can add comment there, that for mocking by other projects it is expected to use the class attribute, to be set before. because it allows you to not change current code, and just add there this change.

Reply:
It can be confusing to overwrite the class attribute...